### PR TITLE
Move main function to events on hui-view for custom views to use

### PR DESCRIPTION
**Pull request by zsarnett**
## Proposed change

Adds events to hui-view that fires Edit, Add, and Delete Card

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/panels/lovelace/views/hui-view.ts [hunk 1]
```diff
@@ -106,6 +118,16 @@ export class HUIView extends UpdatingElement {
 
     if (configChanged && !this._layoutElement) {
       this._layoutElement = createViewElement(viewConfig!);
+      this._layoutElement.addEventListener(
+        "view-create-card",
+        this._showCreateCardDialog
+      );
+      this._layoutElement.addEventListener("view-edit-card", (ev) =>
+        this._showEditCardDialog(ev)
```
Comment (balloob) [inline src/panels/lovelace/views/hui-view.ts:126]: No need to call a function. You can just define it in here. 

We should dynamically import these.

File: src/panels/lovelace/views/hui-view.ts [hunk 2]
```diff
@@ -20,11 +20,23 @@ import { processConfigEntities } from "../common/process-config-entities";
...
+import { confDeleteCard } from "../editor/delete-card";
 import type { Lovelace, LovelaceBadge, LovelaceCard } from "../types";
 
 const DEFAULT_VIEW_LAYOUT = "masonry";
 const PANEL_VIEW_LAYOUT = "panel";
 
+declare global {
+  // for fire event
+  interface HASSDomEvents {
+    "view-create-card": undefined;
```
Comment (balloob) [inline src/panels/lovelace/views/hui-view.ts:34]: The word "view" is very confusing 😆 Sounds like a verb.

Comment (balloob) [reply src/panels/lovelace/views/hui-view.ts:34]: Can we just make it `ll-` instead ?

File: src/panels/lovelace/views/hui-view.ts [hunk 3]
```diff
@@ -228,6 +252,26 @@ export class HUIView extends UpdatingElement {
       curBadgeEl === badgeElToReplace ? newBadgeEl : curBadgeEl
     );
   }
+
+  private async _showCreateCardDialog(): Promise<void> {
```
Comment (balloob) [inline src/panels/lovelace/views/hui-view.ts:256]: inline this

**Review by balloob — APPROVED** [review]

**Review by balloob — APPROVED** [review]
Ok to merge when documentation added 👍

**Comment by zsarnett** [comment]
https://github.com/home-assistant/developers.home-assistant/pull/746

**Pull request merged**